### PR TITLE
Query bundled quotes to get pending address change

### PR DESCRIPTION
**Pull request by hugokallstrom**
### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/ApplicationModule.kt [hunk 1]
```diff
@@ -476,3 +478,10 @@ val useCaseModule = module {
     single { StartNorwegianAuthUseCase(get()) }
     single { SubscribeToAuthStatusUseCase(get()) }
 }
+
+val offerPersistenceManagerModule = module {
+    single {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(get())
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ApplicationModule.kt:484]: Should we let `Koin` instantiate this dependency instead? If we do that, the instance can be reused between dependencies, now we're requiring one to be created specifically for `OfferPersistenceManager` 🤔

File: app/src/main/java/com/hedvig/app/ApplicationModule.kt [hunk 2]
```diff
@@ -476,3 +478,10 @@ val useCaseModule = module {
     single { StartNorwegianAuthUseCase(get()) }
     single { SubscribeToAuthStatusUseCase(get()) }
 }
+
+val offerPersistenceManagerModule = module {
+    single {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ApplicationModule.kt:483]: Should we explicitly mention that we're creating a `single<OfferPersistenceManager>`?

It honestly might not be necessary if this code runs without any issues, I was under the impression that it was needed though with Koin 🤔

File: apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/home.graphql [hunk 3]
```diff
@@ -73,4 +73,23 @@ query HomeQuery($locale: Locale!, $languageCode: String!) {
     }
     body
   }
+
+  quoteBundle(input: $quoteBundleInput) {
```
Comment (hedvigoscar) [inline apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/home.graphql:77]: Does this query gracefully handle being passed an empty list of `ID`s? What about an `ID` that does not exist, will it throw errors? If yes, how do we deal with that error?

File: testdata/src/main/java/com/hedvig/app/testdata/feature/referrals/Data.kt [hunk 4]
```diff
@@ -2,8 +2,10 @@ package com.hedvig.app.testdata.feature.referrals
 
 import com.hedvig.android.owldroid.fragment.MonetaryAmountFragment
 import com.hedvig.android.owldroid.fragment.ReferralFragment
+import com.hedvig.android.owldroid.graphql.HomeQuery
```
Comment (hedvigoscar) [inline testdata/src/main/java/com/hedvig/app/testdata/feature/referrals/Data.kt:5]: Are these imports used? 🤔

**Review by hedvigoscar — COMMENTED** [review]
A few concerns, mostly it's looking pretty good though I think

**Comment by hedvigoscar** [comment]
Also, it looks like there are some build issues that need to be addressed as well 😉

File: apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/home.graphql (hunk 3 above)
Comment (hugokallstrom) [reply apollo/src/main/graphql/com/hedvig/android/owldroid/graphql/home.graphql:77]: You are right, this is not handled very well. At the moment the whole query fails and we get an error message saying we need to supply at least one id. Should we do this query separate from the home query? 

I suppose what we really want is for the query to look something like 
```
Query.quoteBundle(input: QuoteBundleInput!): QuoteBundleResult!
```

Where a `QuoteBundleResult` is a union type and can have values `QuoteBundle` or `EmptyBundle`?

**Pull request closed without merging**